web/packages/extension/src/common.ts

```diff
@@ -32,14 +32,62 @@ class CheckboxOption implements OptionElement<boolean> {
     }
 }
 
+class NumberInputOption implements OptionElement<number | null> {
+    constructor(
+        private readonly numberInput: HTMLInputElement,
+        readonly label: HTMLLabelElement
+    ) {
+        this.numberInput.addEventListener("input", () => {
+            this.numberInput.reportValidity();
+        });
+    }
+
+    get input() {
+        return this.numberInput;
+    }
+
+    get value(): number | null {
+        const ni = this.numberInput;
+        const num = ni.valueAsNumber;
+        if (Number.isNaN(num)) {
+            return null;
+        }
+        if (ni.min) {
+            const min = Number.parseFloat(ni.min);
```

Comment:
`Number.parseFloat()` is not really needed - The more basic `Number(ni.min)` is preferred.